e2e/tests/eth_integration.rs

```diff
@@ -0,0 +1,273 @@
+        Account::Offline(PrivateKey::from_raw(TRADER_BUY_ETH_A_PK).unwrap(), None);
+    let trader_buy_eth_b =
+        Account::Offline(PrivateKey::from_raw(TRADER_BUY_ETH_B_PK).unwrap(), None);
+    let trader_sell_weth =
+        Account::Offline(PrivateKey::from_raw(TRADER_SELL_ETH_PK).unwrap(), None);
+
+    let gpv2 = GPv2::fetch(&web3, &solver).await;
+    let UniswapContracts {
+        uniswap_factory,
+        uniswap_router,
+    } = UniswapContracts::fetch(&web3).await;
+
+    // Create & Mint tokens to trade
+    let token = deploy_mintable_token(&web3).await;
+    tx!(solver, token.mint(solver.address(), to_wei(100_000)));
+    tx!(solver, token.mint(trader_buy_eth_a.address(), to_wei(50)));
+    tx!(solver, token.mint(trader_buy_eth_b.address(), to_wei(50)));
+
+    let weth = WETH9::builder(&web3)
+        .deploy()
+        .await
+        .expect("WETH deployment failed");
+    tx_value!(solver, to_wei(100_000), weth.deposit());
+    tx_value!(solver, to_wei(100), weth.deposit());
+    tx!(
+        solver,
+        weth.transfer(trader_sell_weth.address(), to_wei(100))
+    );
+
+    // Create and fund Uniswap pool
+    tx!(
+        solver,
+        uniswap_factory.create_pair(token.address(), weth.address())
+    );
+    tx!(
+        solver,
+        token.approve(uniswap_router.address(), to_wei(100_000))
+    );
+    tx!(
+        solver,
+        weth.approve(uniswap_router.address(), to_wei(100_000))
+    );
+    tx!(
+        solver,
+        uniswap_router.add_liquidity(
+            token.address(),
+            weth.address(),
+            to_wei(100_000),
+            to_wei(100_000),
+            0_u64.into(),
+            0_u64.into(),
+            solver.address(),
+            U256::max_value(),
+        )
+    );
+
+    // Approve GPv2 for trading
+    tx!(trader_buy_eth_a, token.approve(gpv2.allowance, to_wei(50)));
+    tx!(trader_buy_eth_b, token.approve(gpv2.allowance, to_wei(50)));
+    tx!(trader_sell_weth, weth.approve(gpv2.allowance, to_wei(100)));
+
+    let native_token = weth.address();
+    let OrderbookServices {
+        orderbook,
+        price_estimator,
+    } = OrderbookServices::new(&web3, &gpv2, &uniswap_factory, native_token).await;
+
+    let client = reqwest::Client::new();
+
+    // Test fee endpoint
+    let client_ref = &client;
+    let estimate_fee = |sell_token, buy_token| async move {
+        client_ref
+            .get(&format!(
+                "{}{}?sellToken={:?}&buyToken={:?}&amount={}&kind=sell",
+                API_HOST,
+                FEE_ENDPOINT,
+                sell_token,
+                buy_token,
+                to_wei(42)
+            ))
+            .send()
+            .await
+            .unwrap()
+    };
+    let fee_with_weth = estimate_fee(token.address(), BUY_ETH_ADDRESS).await;
+    assert_eq!(fee_with_weth.status(), 200);
+    // Eth is only supported as the buy token
+    let fee_with_eth = estimate_fee(BUY_ETH_ADDRESS, token.address()).await;
```

Comment:
I think these variable names are off, aren't they?

Reply:
Good find!
At first I wanted to check that querying the two gave the same result, but then I realized that it wouldn't be helpful because the gas price is zero. So now I just compare a buy eth pair to an invalid pair.